### DISCUSSION TITLE
替换功能怎么排序？

### DISCUSSION THREAD

**Discussion by cool-delete** [jing332/tts-server-android#205] [General]
[
    {
        "group": {
            "id": 1694675739055,
            "name": "人名地名官职",
            "order": 3,
            "onExecution": 1
        },
        "list": [
            {
                "id": 1694765722473,
                "groupId": 1694675739055,
                "name": "定难",
                "pattern": "定难",
                "replacement": "定nàn",
                "order": 5
            }
        ]
    },
    {
        "group": {
            "id": 1675488184588,
            "name": "纠正发音【H-Z】",
            "order": 4
        },
        "list": [
            {
                "id": 1720755331778,
                "groupId": 1675488184588,
                "name": "难得",
                "pattern": "难得",
                "replacement": "难得",
                "order": 1
            },
            {
                "id": 1675395291403,
                "groupId": 1675488184588,
                "name": "难nàn",
                "isRegex": true,
                "pattern": "(?<=[患])难\n(?<=患|落|多灾多|磨|责|遭|逃|避|发|罹|遭.{0,3}|经.{0,3}|逃过.)难",
                "replacement": "婻",
                "order": 27
            }
        ]
    },
    {
        "group": {
            "id": 1676833236984,
            "name": "字词_正则",
            "order": 11
        },
        "list": [
            {
                "id": 1676833540496,
                "groupId": 1676833236984,
                "name": "㊣ 难➡️nàn",
                "isRegex": true,
                "pattern": "难(?=民(?!众))|(?<=多灾多|遭.{0,3}|经.{0,3}|逃过.{0,3}|[患落磨责逃避发罹])难",
                "replacement": "婻",
                "order": 26
            },
            {
                "id": 1676834061999,
                "groupId": 1676833236984,
                "name": "㊣ 难➡️nuó",
                "isRegex": true,
                "pattern": "难(?=[戏舞])|(?<=佩玉之)难",
                "replacement": "傩",
                "order": 27
            }
        ]
    }
]

请求音频：岳阳估计这家伙平时可能是一个少爷兵，说不定还是个大家族的子弟，初次上战场，面对危境，能够像现在这样坚持，已经很婻得了。
我明明写了难得的读音，但是变成婻了

**Comment by 550-q** [comment]
规则“㊣ 难➡️nàn”里有：经.{0,3}  
它会把“已经很难得了”的“难”替换成“婻”，这个 “经.{0,3}” 本来是为了保证“经历磨难”的“难”读音不出错的。
如果平时遇不到这个读音错误可以删掉它，也可以新建一个规则，把“很婻得”替换为“很南得”，放在规则“㊣ 难➡️nàn”的下面。
或者把规则“㊣ 难➡️nàn”的替换规则改成这个：
(?!难得)(难(?=民(?!众))|(?<=多灾多|遭.{0,3}|经.{0,3}|逃过.{0,3}|[患落磨责逃避发罹])难)

如果以后还遇到错误把词语里的“难”替换为“婻”的，在替换规则开头括号里的“难得”后面添加那个词就行（注意必须是“难”字在前的词），要注意用“|”隔开。如果是“难”字在后的词，比如“好难”，那么应该是在替换规则最后面添加：(?<!好难)

（注意：你的规则“难nàn”是有错误的，没有起作用，所以这里没有提到需要修改它）
